server/src/transactions/transactions.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { CreateTransactionDto } from './dto/create-transaction.dto';
+import { Transaction } from './entities/transaction.entity';
+import { GetTransactionsFilterDto } from './dto/get-transactions-filter.dto';
+import { TransactionStatus } from './transaction-status.enum';
+import { Repository } from 'typeorm';
+
+@Injectable()
+export class TransactionsService {
+  constructor(
+    @InjectRepository(Transaction)
+    private transactionsRepository: Repository<Transaction>,
+  ) {}
+
+  async createTransaction(
+    createTransactionDto: CreateTransactionDto,
+  ): Promise<Transaction> {
+    return await this.transactionsRepository.save(createTransactionDto);
+  }
+
+  // look at this filter
+  getTransactions(filterDto: GetTransactionsFilterDto): Promise<Transaction[]> {
+    return this.transactionsRepository.find();
+  }
+
+  async getTransactionById(id: number): Promise<Transaction> {
+    // NOTE:  findOne is a built in function
+    const found = await this.transactionsRepository.findOne(id);
+    if (!found) {
+      throw new NotFoundException();
+    }
+    return found;
+  }
+
+  async updateTransaction(
```

Comment:
this should either be `updateTransactionStatus` or be able to update any property
i'm leaning towards the latter unless you see a benefit in having multiple endpoints for updating different properties